src/alg_traits.jl

```diff
@@ -118,6 +118,17 @@ Defaults to false.
 """
 requiresconstraints(opt) = false
 
+"""
+requiresderivative(opt)
+
+Trait declaration for whether an optimizer
+requires derivatives specified in
+`cons` in `OptimizationProblem`.
+
+Defaults to false.
+"""
+requiresderivative(opt) = false
```

Comment:
This is not a very explicit name. Did you mean requires gradient?

Reply:
well we can't really change traits easily. They get set in stone once they are used downstream because it's highly breaking to change base namings. So if we're even considering changing it in the future, we should fix that now.

Reply:
@ParasPuneetSingh you can remove this now, we'll only need the ones below